improver/utilities/cube_manipulation.py

```diff
@@ -295,73 +246,23 @@ def process(self, cubes_in):
         for cube in cubes_in:
             cubes.append(cube.copy())
 
-        # check master coordinate is on cubes - if not, throw error
-        if not all(cube.coords(self.master_coord) for cube in cubes):
-            raise ValueError(
-                "Master coordinate {} is not present on input cube(s)".format(
-                    self.master_coord
-                )
-            )
-
         # slice over requested coordinates
-        if self.coords_to_slice_over is not None:
-            for coord_to_slice_over in self.coords_to_slice_over:
-                cubes = self._slice_over_coordinate(cubes, coord_to_slice_over)
+        if self.coords_to_slice_over:
+            for coord in self.coords_to_slice_over:
+                cubes = self._slice_over_coordinate(cubes, coord)
 
         # remove unmatched attributes
         equalise_cube_attributes(cubes, silent=self.silent_attributes)
 
         # remove cube variable names
         strip_var_names(cubes)
 
-        # promote scalar coordinates to auxiliary as necessary
-        associated_master_cubelist = iris.cube.CubeList([])
-        for cube in cubes:
-            associated_master_cubelist.append(
-                self._associate_any_coordinate_with_master_coordinate(cube)
-            )
-
         # concatenate cube
-        result = associated_master_cubelist.concatenate_cube()
+        result = cubes.concatenate_cube()
 
         return result
 
 
-def concatenate_cubes(
```

Comment:
Looking at the way ConcatenateCubes is called, I think it might actually be better as a solo function. You could then put silent attributes as keyword args and you wouldn't even have any real state missing. What do you think?

Reply:
Possibly a good plan, but I actually think we're going to be able to get rid of ConcatenateCubes completely.  The reason is the only remaining use case is better served by "slice and merge" (rather than "slice, promote, associate aux coords and concatenate"), which could be done very simply in the MergeCubes wrapper.  The reason I haven't done this yet is that it's dependent on unit test input cubes being correct and self consistent - there are some unit tests (notably those for ECC) that I suspect will break if I tried this now, so it would be better to fix them first and see.

Reply:
> I actually think we're going to be able to get rid of ConcatenateCubes completely

Even better! 😄